### DISCUSSION TITLE
Nested Form with an object whose key is a number

### DISCUSSION THREAD

**Discussion by nguyenkhang2801** [react-hook-form/react-hook-form#9409] [Q&A]
I want to use useForm for my form with defaultValue
`defaultValues: {
      example: {
        "0": "ppp",
        "1": "aaa"
      }
    }`
    
But if i use
`<input {...register("example.0")} defaultValue="test" />`
`<input {...register("example.1")} defaultValue="test" />`

It will be change to an array `example: ['ppp', 'aaa']`

Pls help me to resolve this problem, thank you so much!
codeSandbox: [https://codesandbox.io/s/react-hook-form-get-started-ts-forked-gt64m4?file=/src/index.tsx](url)

**Comment by bluebill1049** [comment]
`.integar` is reserved for array type, you will have to do the following instead

`<input {...register("example.index0")} defaultValue="test" />` or convert the array data type back to object at the submission.
